server/core/services.py

```
@@ -0,0 +1,87 @@
+                display_name=data['display_name'],
+            )
+            
+            # Set password (this handles the hashing)
+            user.set_password(data['password'])
+            
+            # Save the user
+            user.save()
+            
+            return user
+        except ValidationError as e:
+            raise ValidationError({'password': e.messages})
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)